internal/service/certificatesigningrequest.go

```diff
@@ -102,6 +122,9 @@ func (h *ServiceHandler) ListCertificateSigningRequests(ctx context.Context, req
 	case nil:
 		return server.ListCertificateSigningRequests200JSONResponse(*result), nil
 	default:
+		if s, ok := selector.IsSelectorError(err); ok {
```

Comment:
Each selector error is a new instance containing the flightctl error that is already allocated and a selector error that contains more information. Since we switch (err) this approach cannot work because we cannot compare.
We can change the switch case to something like switch{ case err ==x: case err==y: , case IsSelectorError(err)}

Reply:
Done, the approach has changed.